src/adapter/src/catalog/storage.rs

```diff
@@ -1840,6 +1844,120 @@ impl<'a> Transaction<'a> {
         Ok(())
     }
 
+    /// Updates cluster `id` in the transaction to `cluster`.
+    ///
+    /// Returns an error if `id` is not found.
+    ///
+    /// Runtime is linear with respect to the total number of items in the stash.
```

Comment:
```suggestion
    /// Runtime is linear with respect to the total number of clusters in the stash.
```

This might be diverging from the existing norm, but I think it'd be clearer.